lib/puppet/provider/package/dpkg.rb

```diff
@@ -144,10 +150,15 @@ def purge
     dpkg "--purge", @resource[:name]
   end
 
-  def hold
+  def deprecated_hold
+    Puppet.warning "\"ensure=>held\" has been deprecated and will be removed in a future version, use \"hold=true\" instead "
```

Comment:
better to use `warning("...")` which is a method mixed into the provider so that the warning message contains context about which resource title/provider has the issue:

```
Warning: Package[something](provider=dpkg): "ensure=>held" has been ...
```